packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1047,6 +1061,57 @@ class EditableText extends StatefulWidget {
   ///    which are more specialized input change notifications.
   final ValueChanged<String>? onChanged;
 
+  /// {@template flutter.widgets.editableText.onContentCommitted}
+  /// Called when a user inserts content through the device keyboard
+  /// on Android only.
+  ///
+  /// The map will contain the following data:
+  ///  - MIME Type (supporting png, bmp, jpg, tiff, gif, jpeg, and webp by default)
+  ///  - Bytes
+  ///  - URI
+  ///
+  /// The bytes represent the content being committed through the keyboard. This can
+  /// be an image, video, or any other file type
+  ///
+  /// {@tool dartpad --template=stateful_widget_material}
+  ///
+  /// This example shows how to access the data for committed content in your
+  /// `TextField`.
+  ///
+  /// ```dart
+  /// final TextEditingController _controller = TextEditingController();
+  ///
+  /// @override
+  /// void dispose() {
+  ///   _controller.dispose();
+  ///   super.dispose();
+  /// }
+  ///
+  /// @override
+  /// Widget build(BuildContext context) {
+  ///   return Scaffold(
+  ///     body: Column(
+  ///       mainAxisAlignment: MainAxisAlignment.center,
+  ///       children: <Widget>[
+  ///         const Text('Here's a text field that supports inserting gif content:'),
+  ///         TextField(
+  ///           controller: _controller,
+  ///           onContentCommitted: (CommittedContent data) async {
+  ///             if (data.mimeType == "image/gif" && data.data != null) {
+  ///               //handle Uint8List (e.g. upload to server, display a MemoryImage, etc)
+  ///               ...
```

Comment:
Using `@tool dartpad` above means that this sample will actually be turned into a live example on the docs website, so this code needs to compile.  You'll have to either comment out this `...` and/or make it a `@tool snippet` to just show the code in a codeblock on the docs site.

Actually this won't work on the web, right?  So it should probably be a `snippet`.

Reply:
Actually, you'll want to use `{@tool sample}` instead of `{@tool snippet}`, since it's a complete example that just doesn't run on the web.  `sample` API doc samples are non-dartdoc full application examples that can be compiled and run on the platforms that support them. `snippet` API doc samples are partial examples that don't compile/run as an application.

Also, you'll want these to be non-inline code without using a template: templates are no longer supported.  Take a look at https://github.com/flutter/flutter/blob/master/examples/api/README.md for more details.

Reply:
Thank you for the clarification!